.github/workflows/build-ci.yml

```diff
@@ -2,8 +2,6 @@ name: CI
 
 on:
     push:
```

Comment:
@divine should this always run on push? I think it makes more sense to only run for a pull request event.

In our application we have the following defined in our code quality Github Workflow. We run on certain pull request event and push to the develop branch. Reduces the Github Action minutes, and double runs of the workflow.

```yml
on:

  pull_request:
    types:
      - opened
      - synchronize
      - ready_for_review

  workflow_dispatch:

  push:
    branches:
      - develop
```

Reply:
As this is an open-source project there is no limit on minutes, so it should be fine.
The reason from what I remember is that sometimes I've pushed commits directly to the branches which of course is unnecessary.
Your suggested change looks great.

Thanks!